### DISCUSSION TITLE
what is the inverse of FrameGenerator

### DISCUSSION THREAD

**Discussion by ghost** [MTG/essentia#1369] [Q&A] [answered]
python essentia standard mode
MonoLoader -> FrameGenerator -> Windowing -> FFT -> IFFT -> ??? -> MonoWriter

**Comment by ghost** [comment]
FrameGenerator converts one-dimensional input to two-dimensional output, so a method is needed to convert two-dimensional input to one-dimensional output. The input and output of OverlapAdd are both one-dimensional, something must be wrong.

**Reply by palonso** [reply, marked as answer]
[OverpadAdd](https://essentia.upf.edu/reference/std_OverlapAdd.html) is the algorithm you are looking for.

The input and output of OverlapAdd are indeed one-dimensional because you are expected to pass it every IFFT frame iteratively. In order to reconstruct the signal you can use, for example, numpy: `np.stack(frames)`:

```python
frames = [overlapAdd(frame) for frame in ifft_frames]
signal = np.stack(frames)
```

Additionally,  have a look at [this](https://github.com/MTG/essentia/blob/master/src/examples/python/musicbricks-tutorials/1-stft_analsynth.py) usage example using OverlapAdd in streaming mode.

**Reply by ghost** [reply]
maybe we need update the document
